File: JsApp/app/core/ruleExecutor.js
function RuleExecutor(){
   var self = this;
   
   self.rules = {};
};

RuleExecutor.prototype.execute = function execute(rule, param){
    var self = this,
        result = null;
        
    if(self.ruleIsValid(rule) && param != null){
        result = self.rules[rule].execute(param);
    }        
        
    return result;
};

RuleExecutor.prototype.addRules = function addRules(rules){
    var self = this;
       
    self.rules = rules;
};

RuleExecutor.prototype.ruleIsValid = function ruleIsValid(rule){
    var self = this,
        result = rule != null && self.rules[rule] != null;
        
    return result;
};

module.exports = RuleExecutor;

File: NetApp/App/App.Core/RuleExecutor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using App.Models; 

namespace App.Core
{
    public class RuleExecutor
    {
        public Dictionary<string, RuleBase> Rules{ get; set; }
        public RuleExecutor()
        {
            this.Rules = new Dictionary<string, RuleBase>();
        }
        public string Execute(string rule, string param)
        {
            string result = string.Empty;

            if (this.Rules.ContainsKey(rule))
            {
                result = this.Rules[rule].Execute(param);
            }

            return result;
        }

        public virtual void AddRule(string rule)
        {
            var ruleObject = new RuleBase(rule);
            this.Rules.Add(rule, ruleObject);
        }
        
    }
}
